linera-ethereum/src/client.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use ethers::{
+    prelude::{Http, Provider},
+    types::{Log, U256},
+};
+use ethers_core::types::{Address, BlockId, BlockNumber, Filter, U64};
+use ethers_middleware::Middleware;
+
+use crate::common::EthereumServiceError;
+
+/// The Ethereum endpoint and its provider used for accessing the ethereum node.
+pub struct EthereumEndpoint {
+    pub url: String,
+    pub provider: Provider<Http>,
+}
+
+impl EthereumEndpoint {
+    /// Connects to an existing Ethereum node and creates an EthereumEndpoint
+    /// if successful.
+    pub fn new(url: String) -> Result<Self, EthereumServiceError> {
+        let provider = Provider::<Http>::try_from(&url)?;
+        let provider = provider.interval(Duration::from_millis(10u64));
+        Ok(Self { url, provider })
+    }
+
+    /// Lists all the accounts of the Ethereum node.
+    pub async fn get_accounts(&self) -> Result<Vec<String>, EthereumServiceError> {
+        Ok(self
+            .provider
+            .get_accounts()
+            .await?
+            .into_iter()
+            .map(|x| format!("{:?}", x))
+            .collect::<Vec<_>>())
+    }
+
+    /// Gets the block number of the Ethereum node.
```

Comment:
Maybe the "highest" or "latest" block number? (I assume that's what it does?)

Reply:
Yes, correct. Thank you.